odoo/addons/base/models/res_lang.py

```diff
@@ -6,17 +6,40 @@
 import locale
 import logging
 import re
-from operator import itemgetter
+from typing import Any, Literal
 
 from odoo import api, fields, models, tools, _
 from odoo.exceptions import UserError, ValidationError
+from odoo.tools import OrderedSet
 
 _logger = logging.getLogger(__name__)
 
 DEFAULT_DATE_FORMAT = '%m/%d/%Y'
 DEFAULT_TIME_FORMAT = '%H:%M:%S'
 
 
+class LangData(tools.frozendict):
+    """ A data class which can access field value like a res.lang record
+    e.g. lang_data.code
+    """
+    __slots__ = ()
+
+    def __getattribute__(self, field: str) -> Any:
+        return self[field]
+
+    def __getitem__(self, field: str) -> Any:
+        return super().get(field, False)
```

Comment:
This makes `lang_data.x` return `False` when `lang_data` is empty (`dummy_lang_data`), but it also returns `False` when the developer makes a typo in `x`, or when `x` is not one of the cached fields.  Shouldn't we make a difference between the empty case and some programming error?

Reply:
It is quite tricky and hard to override `__getattribute__` while making `dict(langdata)` works.
And I have to sacrifice the feature.
But if you think this feature is more important, I change the code.
But I have to override `__getattr__`
And add a warning in the documenation of `CACHED_FIELDS` to ask users don't cache field names which are also dict's method names